metpy/plots/skewt.py

```diff
@@ -329,15 +329,24 @@ def plot(self, p, t, *args, **kwargs):
         """
         # Skew-T logP plotting
         t, p = _delete_masked_points(t, p)
-        lines = self.ax.semilogy(t, p, *args, **kwargs)
 
+        # Set some default y limits if not already provided
+        if self.ax.get_ylim() == (0, 1):
```

Comment:
I wonder if we could use `ax.margins` to get this same behavior?

Reply:
Curious what you mean here. The example I was testing against returns `(0.5, 0.5)` before and after `ax.set_ylim()` is manually called in a script, so I couldn't test against that in the same way, at least from what I can tell.

Reply:
I probably shouldn't have clicked a line. I was wondering if `margins` would handle adding the appropriate padding rather than us doing it manually. I see now, though, that `margins` only works in axes (0, 1) coordinates, not data. So nevermind.

On another note, why are we setting the limits *before* the call to `semilogy`? Doing it after might allow you to use units for the limits.

Reply:
If moved after `semilogy`, some default ylims are set that are data-dependent. Therefore, it's  impossible to know if the user set them or if they are generic. Also since they are data dependent, they're going to be variable, so you can't check for them easily. As a result, I moved the call above that to always catch the default (0,1) ylims, since any variation off that will be user-provided. 

That said, if you know how to get around this, happy to move it back down.